include/zephyr/fs/fs.h

```diff
@@ -87,6 +87,10 @@ enum {
 
 /**
  * @brief File system mount info structure
+ *
+ * For LittleFS, storage_dev must be a flash area ID cast to void*:
+ *     (void *)FIXED_PARTITION_ID(storage)
+ * NOT a pointer to a device.
```

Comment:
I am fine with this, although it should appear above declaration of `storage_dev`.
I was thinking about system specific `fs_mout_t` initializers (functions or macros) with proper checks, but this be ok, for a time being.